# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/ps_split_composite.glsl
# Commit: 0c03e3ad5e56
# Full Hash: 0c03e3ad5e56201e35437734b0ae1d34faa52005
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2020-12-15 09:29:54
# Regressor Bug: 1682365
# File Overlap Count: 1
# Description:
#   Bug 1682365 - Pt 1 - Remove PrevPassAlpha and PrevPassColor r=nical
#   
#   Remove usage of the implicit prev pass alpha and color texture
#   samplers from batching / renderer / shader code. They are replaced
#   by explicit references to the texture ID for the source task.
# ==============================================================================

diff -r 4e0ba157c016 -r 0c03e3ad5e56 gfx/wr/webrender/res/ps_split_composite.glsl
--- a/gfx/wr/webrender/res/ps_split_composite.glsl	Sat Dec 12 22:13:43 2020 +0000
+++ b/gfx/wr/webrender/res/ps_split_composite.glsl	Mon Dec 14 19:38:48 2020 +0000
@@ -86,7 +86,7 @@
 
     gl_Position = uTransform * final_pos;
 
-    vec2 texture_size = vec2(textureSize(sPrevPassColor, 0));
+    vec2 texture_size = vec2(textureSize(sColor0, 0));
     vec2 uv0 = res.uv_rect.p0;
     vec2 uv1 = res.uv_rect.p1;
 
@@ -113,6 +113,6 @@
     float alpha = do_clip();
     float perspective_divisor = mix(gl_FragCoord.w, 1.0, vLayerAndPerspective.y);
     vec2 uv = clamp(vUv * perspective_divisor, vUvSampleBounds.xy, vUvSampleBounds.zw);
-    write_output(alpha * textureLod(sPrevPassColor, vec3(uv, vLayerAndPerspective.x), 0.0));
+    write_output(alpha * textureLod(sColor0, vec3(uv, vLayerAndPerspective.x), 0.0));
 }
 #endif